common/analysis/processingChain.js

```diff
@@ -67,13 +67,18 @@ export const newProcessingChain = (props = {}) => ({
   [Calculation.keys.temporary]: true,
 })
 
-export const newProcessingStep = (processingChain, props = {}) => ({
-  [Step.keys.uuid]: uuidv4(),
-  [Step.keys.processingChainUuid]: getUuid(processingChain),
-  [Step.keys.index]: getProcessingSteps(processingChain).length,
-  [Step.keys.props]: props,
-  [Calculation.keys.temporary]: true,
-})
+export const newProcessingStep = (chain, props = {}) => {
```

Comment:
should we introduce a factory object? e.g. ChainFactory with method createChain, createStep and createCalculation? better for separation of concern

Reply:
right, ok